cli/storage_s3.go

```diff
@@ -25,6 +25,7 @@ func (c *storageS3Flags) Setup(_ StorageProviderServices, cmd *kingpin.CmdClause
 	cmd.Flag("prefix", "Prefix to use for objects in the bucket").StringVar(&c.s3options.Prefix)
 	cmd.Flag("disable-tls", "Disable TLS security (HTTPS)").BoolVar(&c.s3options.DoNotUseTLS)
 	cmd.Flag("disable-tls-verification", "Disable TLS (HTTPS) certificate verification").BoolVar(&c.s3options.DoNotVerifyTLS)
+	cmd.Flag("upload-part-size-bytes", "Custom part size used for uploading the object").Default("0").Uint64Var(&c.s3options.UploadPartSize)
```

Comment:
Can we add some more context in this description? For example:
```suggestion
	cmd.Flag("upload-part-size-bytes", "Size of parts objects are split into for multipart uploading").Default("0").Uint64Var(&c.s3options.UploadPartSize)
```

Reply:
Yes, let's do the UI (https://github.com/kopia/htmlui/blob/main/src/SetupS3.js). This needs to be in separate PR anyway, since the code now lives in a different repository.

Reply:
@jkowalski 

I think the Storj docs I linked inside the issue have a mistake
They reference it as `aws configure set default.s3.multipart_threshold 64MB`

But [AWS cli docs](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#multipart-threshold) say:

> When uploading, downloading, or copying a file, the S3 commands will switch to multipart operations if the file reaches a given size threshold. The multipart_threshold controls this value. You can specify this value in one of two ways:

It should actually be [`multipart_chunksize`](https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#multipart-chunksize)

> Once the S3 commands have decided to use multipart operations, the file is divided into chunks. This configuration option specifies what the chunk size (also referred to as the **part size**) should be. This value can specified using the same semantics as multipart_threshold, that is either as the number of bytes as an integer, or using a size suffix.

Reply:
This is also mentioned as **part size** with [minio docs](https://docs.min.io/docs/golang-client-api-reference.html)
> opts.PartSize	uint64	Specify a custom part size used for uploading the object

Reply:
I did a PR for docs there  https://github.com/storj/docs/pull/192

Reply:
Can anything be done on the client side to **NOT USE** multi-part uploads?

In general, multi-part uploads should be avoided altogether, and kopia is implemented to not require multi-part uploads.

Reply:
@julio-lopez That's the whole idea of this PR.

See it inside [minio client source](https://github.com/minio/minio-go/blob/1a65a8381319e069eee85921b5ffe11289e59a72/api-put-object.go#L257).

If a file (a pack) is smaller than provided `PartSize` or `opts.DisableMultipart` is true then it's uploaded with a single request.
If no `PartSize` is set then `minPartSize` is used which is equal to 16MiB right now.
Note that the default repsository pack size Kopia uses is 20 MiB (`max-pack-size-mb`) so it'll result in 2 parts.

Otherwise see calls to [`OptimalPartInfo`](https://github.com/minio/minio-go/blob/1a65a8381319e069eee85921b5ffe11289e59a72/api-put-object-streaming.go#L256)
It will try using `configuredPartSize` or switch to `minPartSize`. In case of later it'll split it up with this formula:

```go
configuredPartSize = minPartSize // 16MiB = 16777216
partSizeFlt = float64(objectSize / maxPartsCount)  // maxPartsCount is 10000
partSizeFlt = math.Ceil(partSizeFlt/float64(configuredPartSize)) * float64(configuredPartSize)
```

If you calculate it for `objectSize = 20MiB` you'll get `partSizeFlt = 16MiB`.
Note that there are performance benefits to using multiple parts if uploading concurrently, but I'm not sure if thats relevant if individual packs are uploaded in parallel.

Reply:
I see, thanks for the explanation.

What about always setting `opts.DisableMultipart=true` ? Would that work?

Reply:
@julio-lopez that works for me